neqo-transport/src/send_stream.rs

```diff
@@ -1128,10 +1132,10 @@ impl SendStream {
             SendStreamState::Send {
                 ref mut send_buf, ..
             } => {
+                let previous_limit = send_buf.avail();
                 send_buf.mark_as_acked(offset, len);
-                if self.avail() > 0 {
-                    self.conn_events.send_stream_writable(self.stream_id);
-                }
+                let current_limit = send_buf.avail();
+                self.emit_writable_event(previous_limit, current_limit);
```

Comment:
Pull request consolidates the 3 places (`mark_as_acked`, `set_max_stream_data`, `handle_max_data`) where `DataWritable` events are emitted into a single place, i.e. `SendStream::emit_writable_event`.